@Rastaban Thanks for your attention.

I didn't test it with `x86-uwp` or `x64-uwp` before (since I am not familiar with `UWP` platform). But I did give `vcpkg install oniguruma:x86-uwp` a try few minutes ago and found the issue:

>4>e:\vcpkg\buildtrees\oniguruma\src\v6.9.1-752b57c683\src\regparse.c(7366): error C4703: potentially uninitialized local pointer variable 'Then' used [E:\vcpkg\buildtrees\oniguruma\x86-uwp-dbg\onig.vcxproj]
     4>e:\vcpkg\buildtrees\oniguruma\src\v6.9.1-752b57c683\src\regparse.c(6603): error C4703: potentially uninitialized local pointer variable 'tag_end' used [E:\vcpkg\buildtrees\oniguruma\x86-uwp-dbg\onig.vcxproj]
     4>e:\vcpkg\buildtrees\oniguruma\src\v6.9.1-752b57c683\src\regparse.c(6893): error C4703: potentially uninitialized local pointer variable 'tag_end' used [E:\vcpkg\buildtrees\oniguruma\x86-uwp-dbg\onig.vcxproj]

I think it may be caused by the `/W4` compiler option used by `oniguruma`:

https://github.com/kkos/oniguruma/blob/master/CMakeLists.txt#L13-L24

Maybe, we can fix this like this: https://github.com/Microsoft/vcpkg/issues/4577

--- a/ports/oniguruma/CONTROL
+++ b/ports/oniguruma/CONTROL
@@ -1,5 +1,5 @@
 Source: oniguruma
-Version: 6.9.1-2
+Version: 2019-02-26
 Description: Modern and flexible regular expressions library
 
 Feature: non-posix

--- a/ports/oniguruma/portfile.cmake
+++ b/ports/oniguruma/portfile.cmake
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO kkos/oniguruma
-    REF v6.9.1
-    SHA512 a1af020f2042dba2423fb0d0db1ca2c30413c4e950493ac44caa11f616adec98c24df6882413548540095d7f3d8a295afd3ce51e8349ea92fa13d900ee58b126
+    REF 502b1f416746ed8700498229bbfceb180e400fbc
+    SHA512 0faf12f415de59716d8faa4d3dc026874c3bd6a3624f75f2a184843025294eb885d57164ae6dcb916cba5c7d1a4da4bcb0dc23fce3ceae5b34b7320e8f0e2c02
     HEAD_REF master
 )
 

